docs/rules/array-bracket-newline.md

```diff
@@ -0,0 +1,243 @@
+    2
+];
+let e = [
+    function foo() {
+        dosomething();
+    }
+];
+```
+
+### never
+
+Examples of **incorrect** code for this rule with the `"never"` option:
+
+```js
+/*eslint array-bracket-newline: ["error", "never"]*/
+/*eslint-env es6*/
+
+let a = [
+];
+let b = [
+    1
+];
+let c = [
+    1, 2
+];
+let d = [
+    1,
+    2
+];
+let e = [
+    function foo() {
+        dosomething();
+    }
+];
+```
+
+Examples of **correct** code for this rule with the `"never"` option:
+
+```js
+/*eslint array-bracket-newline: ["error", "never"]*/
+/*eslint-env es6*/
+
+let a = [];
+let b = [1];
+let c = [1, 2];
+let d = [1,
+    2];
+let e = [function foo() {
+    dosomething();
+}];
+```
+
+### multiline
+
+Examples of **incorrect** code for this rule with the default `{ "multiline": true }` option:
+
+```js
+/*eslint array-bracket-newline: ["error", { "multiline": true }]*/
+/*eslint-env es6*/
+
+let a = [
+];
+let b = [
+    1
+];
+let c = [
+    1, 2
+];
+let d = [1,
+    2];
+let e = [function foo() {
+    dosomething();
+}];
+```
+
+Examples of **correct** code for this rule with the default `{ "multiline": true }` option:
+
+```js
+/*eslint array-bracket-newline: ["error", { "multiline": true }]*/
+/*eslint-env es6*/
+
+let a = [];
+let b = [1];
+let c = [1, 2];
+let d = [
+    1,
+    2
+];
+let e = [
+    function foo() {
+        dosomething();
+    }
+];
+```
+
+### minItems
+
+Examples of **incorrect** code for this rule with the `{ "minItems": 2 }` option:
+
+```js
+/*eslint array-bracket-newline: ["error", { "minItems": 2 }]*/
+/*eslint-env es6*/
+
+let a = [
+];
+let b = [
+    1
+];
+let c = [1, 2];
+let d = [1,
+    2];
+let e = [
+  function foo() {
+    dosomething();
+  }
+];
+```
+
+Examples of **correct** code for this rule with the `{ "minItems": 2 }` option:
+
+```js
+/*eslint array-bracket-newline: ["error", { "minItems": 2 }]*/
+/*eslint-env es6*/
+
+let a = [];
+let b = [1];
+let c = [
+    1, 2
+];
+let d = [
+    1,
+    2
+];
+let e = [function foo() {
+    dosomething();
+}];
+```
+
+### multiline and minItems
+
+Examples of **incorrect** code for this rule with the `{ "multiline": true, "minItems": 2 }` options:
+
+```js
+/*eslint array-bracket-newline: ["error", { "multiline": true, "minItems": 2 }]*/
+/*eslint-env es6*/
+
+let a = [
+];
+let b = [
+    1
+];
+let c = [1, 2];
+let d = [1,
+    2];
+let e = [function foo() {
+    dosomething();
+}];
+```
+
+Examples of **correct** code for this rule with the `{ "multiline": true, "minItems": 2 }` options:
+
+```js
+/*eslint array-bracket-newline: ["error", { "multiline": true, "minItems": 2 }]*/
+/*eslint-env es6*/
+
+let a = [];
+let b = [1];
+let c = [
+    1, 2
+];
+let d = [
+    1,
+    2
+];
+let e = [
+    function foo() {
+        dosomething();
+    }
+];
+```
+
+
+## When Not To Use It
+
+If you do not want to enforce line breaks after open and before close array brackets
+
+## Related Rules
+
+* [array-bracket-spacing](array-bracket-spacing.md)
```

Comment:
Could you also add a "Compatibility" section that points to the JSCS `validateNewlineAfterArrayElements` rule? We do this for rules that have a JSCS counterpart to help users migrating to ESLint. Example: https://github.com/eslint/eslint/blob/master/docs/rules/object-property-newline.md#compatibility